xa30_workaround/scripts/dcmdat2niix.py

```diff
@@ -159,7 +228,12 @@ def main():
             metadata_copy["ConversionSoftware"] = "dcmdat2niix"
             # save the nifti file
             output_path = output_base.with_suffix(suffix)
-            nib.Nifti1Image(data_array[..., i, :], nifti_img.affine, nifti_img.header).to_filename(output_path)
+            if len(shape) <= 3:
```

Comment:
As I was writing the other comments, it occurred to me it might be cleaner to just add the missing dim to the nifti data, so the rest of the code is the same. Then squeeze it out in the end in this part here. Would make for less conditionals to deal with.

All of this works though so probably fine if you don't feel like changing it.

Reply:
I thought of this, and there's probably some way to make it work, but the nibabel API can be kind of squirrely about adding/removing dimensions to an image, so I decided to leave it as a series of conditional statements.